### PR TITLE
Examples for BAEL-422

### PR DESCRIPTION
**Pull request by NareshBabuPB**
Collection API in Vavr

### CODE REVIEW

File: vavr/src/test/java/com/baeldung/vavr/collections/CollectionAPIUnitTest.java [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+    @Test
+    public void givenQueue_whenEnqueued_thenCorrect() {
+        Queue<Integer> queue = Queue.of(1, 2, 3);
+        Queue<Integer> secondQueue = queue.enqueue(4)
+            .enqueue(5);
+
+        assertEquals(3, queue.size());
+        assertEquals(5, secondQueue.size());
+
+        secondQueue.dequeue()
```
Comment (pivovarit) [inline vavr/src/test/java/com/baeldung/vavr/collections/CollectionAPIUnitTest.java]: Not sure about this structure. Isn't there an easier way of checking if all queue elements are in there? Queue is essentially an iterable so it should be doable easily

Comment (NareshBabuPB) [inline vavr/src/test/java/com/baeldung/vavr/collections/CollectionAPIUnitTest.java]: Yes, Queue is iterable. But, I wanted to show that dequeue() function returns a Tuple2 which is different from a mutable Queue. And that there is a map() function to work on the contents of the Tuple2.

Comment (pivovarit) [inline vavr/src/test/java/com/baeldung/vavr/collections/CollectionAPIUnitTest.java]: This is definitely a must-have to show those things but not this way by nesting assertions in _map_ calls and not by nesting _dequeue_ calls inside _map_ calls.

If you want to show that a Tuple gets returned containing _head_ and _tail_, you can start easily with:
`Tuple2<Integer, Queue<Integer>> result = secondQueue.dequeue();`
 and then perform necessary assertions. 

Next thing, if you want to show a _map_ call, definitely do not go with names like _(k,v)_ which suggest that this key-value relationship. Keep in mind that _map_ always needs to return another _Tuple_ 

`result.map((head, tail) -> Tuple.of(head.toString(), tail))`

There is also a method that you can use for transforming the whole tuple into something that is not wrapped in the Tuple:
`result.apply((head, tail) -> head.toString());`

and so on... you can store results of those operations as variables and perform assertions on them.

File: vavr/src/test/java/com/baeldung/vavr/collections/CollectionAPIUnitTest.java [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+        javaList.add(4);
+    }
+
+    @Test
+    public void givenList_whenSquared_thenCorrect() {
+        List<Integer> vavrList = List.of(1, 2, 3);
+        Number sum = vavrList.map(i -> i * i)
+            .sum();
+
+        assertEquals(new Long(14), sum);
```
Comment (pivovarit) [inline vavr/src/test/java/com/baeldung/vavr/collections/CollectionAPIUnitTest.java]: The boxing of Long is unnecessary

File: vavr/src/test/java/com/baeldung/vavr/collections/CollectionAPIUnitTest.java [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+
+        Iterator<String> iterator = set.iterator();
+        assertEquals("Blue", iterator.next());
+        assertEquals("Green", iterator.next());
+        assertEquals("Red", iterator.next());
+    }
+
+    @Test
+    public void givenSortedSet_whenReversed_thenCorrect() {
+        Comparator<String> reverseCompare = (a, b) -> b.compareTo(a);
```
Comment (pivovarit) [inline vavr/src/test/java/com/baeldung/vavr/collections/CollectionAPIUnitTest.java]: the `(a, b) -> b.compareTo(a)` can be replaced with `Comparator.reverseOrder();`

File: vavr/src/test/java/com/baeldung/vavr/collections/CollectionAPIUnitTest.java [hunk 4]
```diff
@@ -0,0 +1,261 @@
...
+        assertEquals(new Integer(3), iterator.next()
+            ._1());
+    }
+
+    @Test
+    public void givenJavaList_whenConverted_thenCorrect() {
+        java.util.List<Integer> javaList = java.util.Arrays.asList(1, 2, 3, 4);
+        List<Integer> vavrList = List.ofAll(javaList);
+
+        assertTrue(vavrList instanceof io.vavr.collection.List);
```
Comment (pivovarit) [inline vavr/src/test/java/com/baeldung/vavr/collections/CollectionAPIUnitTest.java]: Those _instanceof_ assertions do not really bring anything new here. You can easily remove them and leave this test without assertions. After all, those are just supposed to be a short `main` methods and not full-blown tests

File: vavr/src/test/java/com/baeldung/vavr/collections/CollectionAPIUnitTest.java [hunk 5]
```diff
@@ -0,0 +1,261 @@
...
+        Iterator<String> iterator = set.iterator();
+        assertEquals("Red", iterator.next());
+        assertEquals("Green", iterator.next());
+        assertEquals("Blue", iterator.next());
+    }
+
+    @Test
+    public void givenMap_whenIterated_thenCorrect() {
+        Map<Integer, List<Integer>> map = List.rangeClosed(0, 10)
+            .groupBy(i -> i % 2);
```
Comment (pivovarit) [inline vavr/src/test/java/com/baeldung/vavr/collections/CollectionAPIUnitTest.java]: Please use the provided formatter. Continuation indents should be 2 spaces

File: vavr/src/test/java/com/baeldung/vavr/collections/CollectionAPIUnitTest.java (hunk 4 above)
Comment (NareshBabuPB) [inline vavr/src/test/java/com/baeldung/vavr/collections/CollectionAPIUnitTest.java]: Is it Ok to have this example in a private method in the same test class? Just wondering if a separate main class is required just for a few lines of code.

For rest of the comments, will work on it as suggested.

**Pull request merged**